tests/dicts/keypath/test_keypath_dict_list_wildcard.py

```diff
@@ -4,20 +4,321 @@
+                        "c": 2,
+                        "d": [4, 5, 6, [0]],
+                    },
+                    {
+                        "c": 3,
+                        "d": [7, 8, 9, [0]],
+                    },
+                ],
+            },
+        }
+        b = KeypathDict(d)
+
+        self.assertEqual(b.get("a.b[0].d[*]"), [1, 2, 3, [0]])
+        self.assertEqual(b.get("a.b[1].d[*]"), [4, 5, 6, [0]])
+        self.assertEqual(b.get("a.b[2].d[*]"), [7, 8, 9, [0]])
+
+        self.assertEqual(
+            b["a.b[*]"],
+            [
+                {
+                    "c": 1,
+                    "d": [1, 2, 3, [0]],
+                },
+                {
+                    "c": 2,
+                    "d": [4, 5, 6, [0]],
+                },
+                {
+                    "c": 3,
+                    "d": [7, 8, 9, [0]],
+                },
+            ],
+        )
+        self.assertEqual(b.get("a.b[2].d[3][*]"), [0])
+
+    def test_wildcard_list_indexes_with_quotes(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+        self.assertEqual(b.get("a['*']"), [1, 2, 3])
+        self.assertEqual(b.get('a["*"]'), [1, 2, 3])
+
+    def test_wildcard_pop_with_flat_list(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+
+        self.assertEqual(b.pop("a[*]"), [1, 2, 3])
+        self.assertEqual(b, {})
+
+    def test_wildcard_pop_with_flat_list_and_default(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+
+        with self.assertRaises(KeyError):
+            b.pop("b[*]")
+        self.assertEqual(b.pop("b[*]", 4), 4)
+
+    def test_wildcard_pop_with_wrong_index(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+        with self.assertRaises(KeyError):
+            b.pop("b[*]")
+
+    def test_wildcard_pop_with_wrong_index_and_default(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+        self.assertEqual(b.pop("b[*]", 6), 6)
+
+    def test_wildcard_pop_with_nested_list(self):
+        d = {
+            "a": {
+                "b": [
+                    {
+                        "c": 1,
+                        "d": [1, 2, 3, [0]],
+                    },
+                    {
+                        "c": 2,
+                        "d": [4, 5, 6, [0]],
+                    },
+                ],
+            },
+        }
+        b = KeypathDict(d)
+
+        self.assertEqual(b.pop("a.b[0].d[-1][*]"), [0])
+        self.assertEqual(b["a.b[0].d"], [1, 2, 3])
+
+        self.assertEqual(b.pop("a.b[0].d[*]"), [1, 2, 3])
+        self.assertEqual(b["a.b[0]"], {"c": 1})
+
+        self.assertEqual(b.pop("a.b[0]"), {"c": 1})
+        self.assertEqual(
+            b["a.b[0]"],
+            {
+                "c": 2,
+                "d": [4, 5, 6, [0]],
+            },
+        )
```

Comment:
Would be interesting also test multiple consecutive wildcards in the same keypath (list of lists):
```python
d['l.[*].[*]']
```

Reply:
I would say correct syntax is as following:
```python
d['l[*][*]']
```

I will push some tests.

Reply:
Yes, this syntax would be more correct, it should be tested with normal indexes too, because as I remember the keys are always splitted using the `keypath_separator`, so using this syntax could create some unexpected problems.